### PR TITLE
[WIP] Numpy format fixes

### PR DESCRIPTION
**Pull request by AlexDBlack**
Working towards fixing: https://github.com/deeplearning4j/deeplearning4j/issues/7466
(Not yet fixed)

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/serde/NumpyFormatTests.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+public class NumpyFormatTests extends BaseNd4jTest {
+
+    public NumpyFormatTests(Nd4jBackend backend) {
+        super(backend);
+    }
+
+    @Test
+    public void testToNpyFormat() throws Exception {
+
+        File dir = new File("C:\\DL4J\\Git\\dl4j-test-resources\\src\\main\\resources\\numpy_arrays");
```
Comment (agibsonccc) [inline nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/serde/NumpyFormatTests.java]: Classpath resource at some point? :)

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-native-api/src/main/java/org/nd4j/nativeblas/BaseNativeNDArrayFactory.java [hunk 2]
```diff
@@ -78,14 +78,9 @@ public Pointer convertToNumpy(INDArray array) {
         header.capacity(size.get());
         header.position(0);
 
-        char[] magic = {'\\','x','9','3','N','U','M','P','Y','1','0'};
-
-        BytePointer magicPointer = new BytePointer(new String(magic).getBytes());
-        BytePointer bytePointer = new BytePointer(magicPointer.capacity() + (int) (size.get() + (array.data().getElementSize() * array.data().length())));
+        BytePointer bytePointer = new BytePointer((int) (size.get() + (array.data().getElementSize() * array.data().length())));
```
Comment (agibsonccc) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-native-api/src/main/java/org/nd4j/nativeblas/BaseNativeNDArrayFactory.java]: Have we looked at the numpy versios to make sure the header hasn't changed?

**Comment by AlexDBlack** [comment]
@agibsonccc this is nowhere near ready for review or merging.
At this point it's merely a starting point for handing off to c++ guys for further work - https://github.com/deeplearning4j/deeplearning4j/issues/7466#issuecomment-481667151

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-native-api/src/main/java/org/nd4j/nativeblas/BaseNativeNDArrayFactory.java (hunk 2 above)
Comment (raver119) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-native-api/src/main/java/org/nd4j/nativeblas/BaseNativeNDArrayFactory.java]: Yes, that wasn't the case.

**Pull request merged**